### DISCUSSION TITLE
Boot on Startup of Headless Raspberry Pi / Media Player Fix in V3.08 Release Notes

### DISCUSSION THREAD

**Discussion by DoogieHowitzerWMD** [GerryDazoo/Slinger#142] [Q&A] [answered]
Hey Gerry,

First off, thank you for your contributions with this. It's really awesome and it's helping out a lot of people, myself included.

I have a couple different questions... they aren't specifically tied together, so I suppose I'll just list them out.

- **While running the Sling Server on a headless Raspberry Pi, what would be needed to start the server on boot?**
&nbsp;
Because this is being run as a service (the copy of sling.service we put in the "systemd" folder), I'm thinking that maybe some parameter within the file itself would need to be modified. I am trying to learn how the different lines and definitions in a service file work, but I could unknowingly be getting in some very deep waters here. Plus, having it here might help others down the line.
For reference (should any of these particulars alter a solution's methodology), at boot, I am brought to the command line, where I have my default user (who has the config, remote, and server.py files in it's home directory) automatically logged in without a password. Once I have this running comfortably, I will eventually require password input for "sudo" commands, just so I don't have an open login with root access.

- **In the release notes for V3.08, you mention a fix for a media player that was shutting down the stream after receiving an OK message - was that for VLC?**
&nbsp;
If so, thank you. I was having an issue with that when trying to open an Http stream with VLC on my PC (on same network as device running Sling Server), but VLC on my phone (android) had not been giving me this issue, whether on the wifi of the local network or outside of the network.
If not, then I'll report here that that is currently an issue I am experiencing. When I have a chance to, I can upload the log files for that, as I had seen some attempt at connection that was successful, but then seems interrupted from any further messages you see with a stable connection.

**Comment by GerryDazoo** [comment, marked as answer]
If you install the service per the instructions in the readme it should restart on boot up.

The media player in question was PotPlayer. Yeah, I never heard of it either but somebody was using it and had that issue.

I don't have any issues with VLC please make sure you have the latest and greatest. My software doesn't know anything about the version of VLC or any other media player connecting, so if one version works and another doesn't it's something on the client side that I can do nothing about.

**Reply by DoogieHowitzerWMD** [reply]
> If you install the service per the instructions in the readme it should restart on boot up.
> 
> The media player in question was PotPlayer. Yeah, I never heard of it either but somebody was using it and had that issue.
> 
> I don't have any issues with VLC please make sure you have the latest and greatest. My software doesn't know anything about the version of VLC or any other media player connecting, so if one version works and another doesn't it's something on the client side that I can do nothing about.

Huh, I hadn't realized it already should be starting on boot already. I was always initiating it with a command after I had logged in.

Thanks, Gerry. I'll try booting my Pi and connecting to a stream without re-entering the start command in the terminal. I'll report back if I have any trouble with that.

&nbsp;

I'll take a look at VLC on my PC again. I should have the latest version already. I believe there are log files associated with it, so I might have to dive into those. Either that, or just use a different media player on my desktop for the time being.

**Reply by DoogieHowitzerWMD** [reply]
Can confirm the server is already running at boot up. I can start a stream as soon as the system is up.
